### PR TITLE
Trust corefinement more (fenced under feature) for even more speedups

### PR DESCRIPTION
**Pull request by ochafik**
**TL;DR: Up to 3x faster `fast-csg` by enabling `fast-csg-trust-corefinement` (might get a couple of crashes, please report with the .off files generated when `--enable=fast-csg-debug-corefinement` is passed in CLI).**

In the early days of https://github.com/openscad/openscad/pull/3637 I was optimistic for 10x to 100x improvements with corefinement operations instead of Nef polyhedron (see these [old benchmarks](https://gist.github.com/ochafik/2db96400e3c1f73558fcede990b8a355)). But the `fast-csg` feature that’s just landed in Nightly builds (https://github.com/openscad/openscad/pull/4087), only realistically bring 3x to 10x speedups. The difference is because the corefinement library can be finicky with the inputs it gets, and can throw / at times corrupt the mesh when it’s given bad inputs. I’ve added checks to avoid those error cases proactively but they're so expensive some files are just 3x slower.

So… this PR adds two experimental features:
*   `fast-csg-trust-corefinement`: an override that says “yeah go ahead and use refinement without any checks, let’s see if it crashes & try to recover gracefully 🤞” 
*   `fast-csg-debug-corefinement`: this will create one `.off` file per operand of each corefinement operation (in the working directory, which... is unclear what is is on MacOS X unless you use the CLI). If a user's model crashes and they're willing to share the generated OFFending files, we can use them to file bugs to CGAL and/or find lighter ways to prevent the bad cases from being sent to corefinement routines.

### CODE REVIEW

**Review by t-paul — CHANGES_REQUESTED** [review]
Code change looks good but it seems like some build files got included.

File: build_integ_dbg/tests/CTestCustom.cmake [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Generated by cmake from /Users/ochafik/Documents/GitHub/openscad/tests/CTestCustom.template
```
[the remark this answers is not in the document]
Comment (ochafik) [reply build_integ_dbg/tests/CTestCustom.cmake]: oops, removed

[the remark this answers is not in the document]
Comment (t-paul) [reply build_integ_dbg/tests/CTestCustom.cmake]: There's a second one (same folder), I could not click as github decided it's too big. Can you drop that too please. Feel free to just rebase for dropping them from history too.

**Review by t-paul — APPROVED** [review]
